Source/Modules/scilab.cxx

```diff
@@ -938,33 +874,51 @@ class SCILAB:public Language {
     for (int i = 0; i < Len(sourceFileList); i++) {
       String *sourceFile = Getitem(sourceFileList, i);
       if (i == 0) {
-	Printf(builderCode, "files = \"%s\";\n", sourceFile);
+	      Printf(builderCode, "files = \"%s\";\n", sourceFile);
       } else {
-	Printf(builderCode, "files($ + 1) = \"%s\";\n", sourceFile);
+	      Printf(builderCode, "files($ + 1) = \"%s\";\n", sourceFile);
       }
     }
 
-    Printf(builderCode, "table = [");
+    Printf(builderCode5, "table = [");
+    Printf(builderCode6, "table = [");
   }
 
   /* -----------------------------------------------------------------------
    * addFunctionInBuilderCode()
    * Add a function wrapper in the function table of generated builder script
    * ----------------------------------------------------------------------- */
 
-  void addFunctionInScriptTable(const_String_or_char_ptr scilabFunctionName, const_String_or_char_ptr wrapperFunctionName, String *scriptCode) {
+  void addFunctionInScriptTable(const_String_or_char_ptr scilabFunctionName, const_String_or_char_ptr scilabSmallFunctionName, const_String_or_char_ptr wrapperFunctionName, String *scriptCode5, String *scriptCode6) {
     if (++builderFunctionCount % 10 == 0) {
-      Printf(scriptCode, "];\ntable = [table;");
+      Printf(scriptCode5, "];\ntable = [table;");
+      Printf(scriptCode6, "];\ntable = [table;");
     }
-    Printf(scriptCode, "\"%s\",\"%s\";", scilabFunctionName, wrapperFunctionName);
+    Printf(scriptCode5, "\"%s\",\"%s\";", scilabSmallFunctionName, wrapperFunctionName);
+    Printf(scriptCode6, "\"%s\",\"%s\";", scilabFunctionName, wrapperFunctionName);
   }
 
   /* -----------------------------------------------------------------------
    * saveBuilderFile()
    * ----------------------------------------------------------------------- */
 
   void saveBuilderFile(String *gatewayName) {
-    Printf(builderCode, "];\n");
+    Printf(builderCode5, "];\n");
+    Printf(builderCode6, "];\n");
+
+    Printf(builderCode, "ver = getversion('scilab');\n");
+    Printf(builderCode, "if ver(1) <= 5 & ver(2) <= 5 & ver(3) <= 2 then\n");
```

Comment:
This check looks wrong to me - for example, 5.4.3 and 4.6.0 would be treated as "6.0.0 or more" (perhaps those aren't actually scilab releases, but my point is that the logic of the check is wrong).

Why not just test `if ver(1) < 6 then` here?

Reply:
you are correct, this seems a logic bug